app/src/webview/java/org/mozilla/focus/webview/HttpAuthenticationDialog.java

```diff
@@ -0,0 +1,112 @@
+package org.mozilla.focus.webview;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.view.KeyEvent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.view.inputmethod.EditorInfo;
+import android.widget.TextView;
+import android.widget.TextView.OnEditorActionListener;
+import org.mozilla.focus.R;
+
+public class HttpAuthenticationDialog {
+
+    private final Context mContext;
+
+    private final String mHost;
+    private final String mRealm;
+
+    private AlertDialog mDialog;
+    private TextView mUsernameView;
+    private TextView mPasswordView;
+
+    private OkListener mOkListener;
+    private CancelListener mCancelListener;
+
+
+    public HttpAuthenticationDialog(Context context, String host, String realm) {
+        mContext = context;
+        mHost = host;
+        mRealm = realm;
+        createDialog();
```

Comment:
Creating the dialog from the constructor is not very clean. I'd expect an API more like:

Builder()
 .set..
 .set..
 .build()

Almost like AlertDialog.Builder but for this purpose here.